### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by LUKE2037** [shion-app/shion#2] [Ideas]
First of all, do I need to leave a message here in Chinese or English. I am a student in the Department of Industrial Design, and usually we also have some knowledge related to UI/UX. There are some questions.

It is not a good idea to hide the menu options on this page in the navigation bar. People usually believe that the navigation bar only serves as a table of contents and should not be used for related operations on the page. If there are difficulties in modifying the program, making this particular button more prominent than other icons is a solution.
Using card vibration to express permission for drag and drop can make people feel uncomfortable, but this is a small issue, everyone can understand what you mean.
The use of software is quite complex, it is recommended to include detailed operating instructions. If users are unable to use your software smoothly, even hard written programs cannot reflect their value.
The most important issue is that the monitoring of software usage time is very inaccurate. It's a hassle for users to keep track of their schedules themselves. If Shion can accurately monitor the duration of software usage, it would be a great help! Can you explain the working principle of the monitoring function?

Finally, I wish developers smooth development and software success. 
-LUKE2037

**Comment by hanaTsuk1** [comment]
First of all, thank you for your use and feedback

> It is not a good idea to hide the menu options on this page in the navigation bar.

I really don't know much about UI/UX. I referred to some [material design specifications](https://m3.material.io/components/navigation-rail/guidelines) for the menu buttons in the navigation bar. Not prominent enough, this is indeed a problem
If you have any other suggestions, you can let me know

> The use of software is quite complex, it is recommended to include detailed operating instructions. If users are unable to use your software smoothly, even hard written programs cannot reflect their value.

Yes, I will add instructions for use. No one used or provided feedback before, and I'm too lazy to write about it🤣

> The most important issue is that the monitoring of software usage time is very inaccurate. 

I have been using it for several months, and this situation should not occur. If you are saying that the timeline data is not accurate enough, you can adjust the `behavior` in the `setting`

> Can you explain the working principle of the monitoring function?

1. When switching to a window, it will be recorded. If no operation is performed on this window (mouse, keyboard) within 1 minute, this record ends.
2. When switching to a window, the recording of the previous window ends
3. The window in which the audio is being played is only related to the active state of the audio. If the audio stops playing, follow rules 1 and 2

I hope you can understand what I mean. I'm not very good at English

**Comment by LUKE2037** [comment]
所以大兄弟咱能用中文不，我开头就问这个讨论区可以不可以用中文🤣但是忘记打问号了哈哈
看shion官网只有中英两种文本我就猜开发者是不是国人了

谷歌的开发标准我也没仔细看过，所以我对于具体的UI设计并不十分清楚，我们只学基本理论。以下是一些相关的想法：
1.点击刷新按钮后没有任何反馈，我不知道我的操作是否成功了。通常我们会给出刷新成功的提示或者一个圆圈转动的小动画来表示这次按钮操作是成功的。让用户了解到shion的工作状态十分重要，这能够提升用户的掌控感和安全感。不然可能会使人疑惑。
![image](https://github.com/shion-app/shion/assets/155209864/15f9a0d0-e7b6-4b3b-aa86-7abb5eee65d1)
2.页面内的操作最好放在页面内，这个按钮放在导航栏不符合逻辑。而且没有其他软件使用过这个按钮，用户很难迅速地明白这个按钮的作用或者页面的操作方法。软件操作逻辑和其他软件相同是最好的，这能很大程度上降低使用门槛和学习成本。“通过通过颤抖的形式来表示卡片可以被移动”也和这个原则有关，其他软件很少见到这种表示方法，所以看起来可能会有点令人疑惑，建议模仿手机桌面app摆放的操作逻辑。
![image](https://github.com/shion-app/shion/assets/155209864/ff8c1df0-79a8-42fb-8247-6c095357b75a)
3.非常美观直观的展示效果！但是我很难自己摸索出这种展示效果的制作方法。图表有哪些种类？不同种类的图表对应了什么信息？标签看起来是和计划有关的，那么我如何在软件内了解到它们之间的关系？
用户在使用软件时会建立一个心理模型来描述软件的运行方式，比如我知道计划页面可以建立计划卡片，我需要点击计划页面的图标然后再点击左下角的万能按钮并设置参数。以建立一个计划卡片为目标，让用户知道应该怎么做才能达成目标，并且知道按下这个按钮之后会发生什么事，这就是心理模型的作用。shion可能需要一份详细的使用说明来帮助用户建立心理模型，说明各个板块功能的作用和联系，这样用户才能够轻松并高效率的使用shion。
![image](https://github.com/shion-app/shion/assets/155209864/ea055215-e409-4e8d-b604-21a1f75e573f)
![image](https://github.com/shion-app/shion/assets/155209864/5b15328e-028b-4ce2-a98c-2c7a9ea39240)

一些其他问题：
4.点确定点不动，没有显示问题出在哪，滑了一下才发现下面的数据类型没有填。建议滚轮自动滑到未填项目让用户能看见，或者跳出提示xx项未填也行。
![image](https://github.com/shion-app/shion/assets/155209864/846a9ffa-1c2b-4f74-a01d-a763b99b9a3a)
![image](https://github.com/shion-app/shion/assets/155209864/93e8711c-8d30-4735-93cf-9a1b2e333ef3)
5.无法添加我想要监控的程序，比如Rainbow6🤣，这一部分的工作原理是啥？能不能让我自己去任务管理器里面找进程监控？
![image](https://github.com/shion-app/shion/assets/155209864/98431367-e424-4ee2-ba4f-003ea0e698b5)
6.我首先建立了一个1*1的组件，然后把它改成了3*1，再然后我拖拽解锁重新排列各个组件，这个组件一旦移动就会变回1*1的大小
![image](https://github.com/shion-app/shion/assets/155209864/5348ad14-47ee-46f2-a942-712cbb90588b)
![image](https://github.com/shion-app/shion/assets/155209864/b400ed6f-9f6d-48b0-99f3-8c965d767bcb)
![image](https://github.com/shion-app/shion/assets/155209864/eaec6904-6c6a-4b91-92fe-fd610298b64b)
![image](https://github.com/shion-app/shion/assets/155209864/d2042065-d614-4847-a0fe-fb35ee2106b5)

**Comment by hanaTsuk1** [comment]
1. 我会添加一个刷新操作动画的
2. 菜单这个位置我还是保留，但是加上明显区别于导航按钮的标识。颤抖的形式表示移动因为ios就是这么做的，我还以为挺常见的，你有什么建议吗？
  2.1. 在内容添加一个加号块用于操作这个设计我见过，但说实话我不是很喜欢，因为这个操作的按钮和数据展示排在一起
3. 这一点我会在近几天添加说明文档的
4. 这个会改进的
5. 这一点我也会加入说明文档的，这里获取的程序列表就是你已经打开了的程序，因为直接给个文件管理器让用户找，不一定能找到正确的入口
6. 可能是个bug，之后修复，给一个复现的步骤吧

还有一点，你之前提到的数据不准确，用了设置有效果了吗？

**Comment by LUKE2037** [comment]
2.我只用过安卓的所以下意识觉得很怪，不过就像我第一次回复一样这并不是什么大问题可以不改，或者来个ab测。
5.重新测试发现还是不能获取到rainbow6的窗口，可能是反作弊手段？想用来控制自己玩游戏的时间来着。
![Snipaste_2024-03-09_11-49-26](https://github.com/shion-app/shion/assets/155209864/1853ffbf-c19d-4f90-b319-2bca8b93b397)
6.复现步骤已在第二次回复中说明。还需要更详细的吗？

监听数据方面我没有调整设置，我现在相信这个功能运行良好。之前的疑惑主要是因为刷新后没有反馈，我怀疑功能出错，加上动画后应该就不会有这种怀疑了。在你的用户文档和程序进行过一次更新后我会再来反馈。shion让我感觉是一款小而美的软件，在工作学习的情境下它似乎能给予很大帮助。期待你的下一次更新，加油！

**Comment by hanaTsuk1** [comment]
5. 你试一下"以管理员身份运行"看能不能找到
6. 不用了

👀

**Comment by LUKE2037** [comment]
5.还是不行，测试了彩六和杀戮尖塔捕捉不到但是beholder和传送门2可以（所以能不能留一个拓展按钮让用户可以从任务管理器里面选择进程监听）

7.我会用浏览器看学习视频，也会用来娱乐，能否增加功能使用户可以自己标记网站分类计算使用时长。不算一个紧迫的需求，程序实现上可能比较复杂？可以放到后面的版本更新中。

**Comment by hanaTsuk1** [comment]
5.首先我要解释一下，现在获取程序列表的方式，就是你想的任务管理器中展示的那些前台程序，至于为什么有些没有获取到，现在还不清楚。你说的这些软件我都没有，所以不能复现找出原因，需要等我测试一下其他软件
5.1. 我需要知道这两个软件和其他有什么不同，另外如果你有兴趣搞清楚原因的话，了解一下powershell，这里是获取列表的[脚本](https://github.com/shion-app/watcher/blob/main/powershell/get-program.ps1)，把第47行删除掉后，运行就能获取到列表
5.2. 添加另一种获取方式可能也不错，但能用一种解决就不用多引入一种

7.这个功能不考虑，我在寻找一种通用的从画面中提取信息的手段用于自动监听

**Comment by hanaTsuk1** [comment]
@LUKE2037 除了程序列表获取，其余修改已发布于[v0.8.6](https://github.com/shion-app/shion/releases/tag/v0.8.6)

**Comment by xiewencan** [comment]
你这个卸载貌似有有点问题，
![屏幕截图 2024-04-26 124014](https://github.com/shion-app/shion/assets/108505125/0b27f07b-6cbd-4ac1-8a90-18bfef40cd74)
他貌似把我这个文件内的东西全删了

**Reply by hanaTsuk1** [reply]
你在安装的时候没有给shion创建一个单独的文件夹   而是使用的路径"E:\应用下载\PCManager"是吗？
如果是的话  卸载软件会删除安装路径下的全部文件是正常情况

**Reply by xiewencan** [reply]
是的，所以我建议你可以让其自动创建一个文件夹，因为存在一些用户的习惯可能没有这么好。当然，经历过后，我已经养成了此习惯。感想你的回答。

**Reply by hanaTsuk1** [reply]
tauri已对此情况进行改进，我将在新版本跟进

**Reply by xiewencan** [reply]
好的
